### PR TITLE
Document process for deleting a deployment (#1054)

### PR DESCRIPTION
**Pull request by jessebrennan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,4 +46,14 @@ check_trufflehog:
...
+	$(MAKE) unsubscribe
+	$(MAKE) -C terraform destroy
+	$(MAKE) -C lambdas delete
+
+force_demolish: check_branch_personal
+	$(MAKE) unsubscribe
+	$(MAKE) -C terraform auto_destroy
+	$(MAKE) -C lambdas delete
+
+.PHONY: all hello terraform deploy subscribe everything reindex clean test travis integration_test trufflehog check_trufflehog delete force_delete
```
Comment (GPelayo) [inline Makefile]: You need to add `demolish ` and `force_demolish ` phony targets. And is `force_delete` a target and should it still be in `PHONY`?

Comment (jessebrennan) [reply Makefile]: Thanks, good catch.

File: scripts/check_branch.py [hunk 2]
```diff
@@ -74,7 +79,10 @@ def main(argv):
...
         stage = config.deployment_stage
         check_branch(branch, stage)
+    if args.personal:
+        if config.deployment_stage in config.main_deployments_by_branch.values():
+            raise RuntimeError(f"Selected deployment '{stage}' is not a personal deployment.")
 
 
 if __name__ == "__main__":
-    main(sys.argv[1:])
+     main(sys.argv[1:])
```
Comment (hannes-ucsc) [inline scripts/check_branch.py]:
```suggestion
    main(sys.argv[1:])
```

File: lambdas/Makefile [hunk 3]
```diff
@@ -12,4 +12,8 @@ clean:
 	$(MAKE) -C indexer clean
 	$(MAKE) -C service clean
 
-.PHONY: all indexer service clean
+delete:
```
Comment (hannes-ucsc) [inline lambdas/Makefile]: This can stay, too.

File: Makefile [hunk 4]
```diff
@@ -46,4 +46,14 @@ check_trufflehog:
 trufflehog: check_trufflehog
 	trufflehog --regex --rules .trufflehog.json --entropy=False file:///$$azul_home
 
-.PHONY: all hello terraform deploy subscribe everything reindex clean test travis integration_test trufflehog check_trufflehog
+demolish: check_branch_personal
```
Comment (hannes-ucsc) [inline Makefile]: Remove both. My apologies for wavering on this.

Comment (jessebrennan) [reply Makefile]: Does that mean also remove the changes to the check_branch script?

**Comment by jessebrennan** [comment]
@hannes-ucsc I restructured the commits to better reflect the changes now.

**Pull request merged**